### ISSUE TITLE
segfault when --opt-file is used

### ISSUE THREAD

**Issue by honggyukim** [namhyung/uftrace#709]
Here is an example
```
$ cat opts
--no-pager -D 3 t-abc

$ uftrace --opt-file opts
# DURATION     TID     FUNCTION
   2.193 us [ 56595] | __monstartup();
   1.176 us [ 56595] | __cxa_atexit();
            [ 56595] | main() {
            [ 56595] |   a() {
   2.137 us [ 56595] |     b();
   3.663 us [ 56595] |   } /* a */
   4.653 us [ 56595] | } /* main */
WARN: Segmentation fault
Segmentation fault (core dumped)
```
Here is address sanitizer report
```
$ ./uftrace -L. --opt-file opts
# DURATION     TID     FUNCTION
   2.086 us [ 57331] | __monstartup();
   1.160 us [ 57331] | __cxa_atexit();
            [ 57331] | main() {
            [ 57331] |   a() {
   2.057 us [ 57331] |     b();
   3.387 us [ 57331] |   } /* a */
   4.490 us [ 57331] | } /* main */
=================================================================
==57330==ERROR: AddressSanitizer: attempting free on address which was not malloc()-ed: 0x60300000efc0 in thread T0
    #0 0x7f06f72f532a in __interceptor_free (/usr/lib/x86_64-linux-gnu/libasan.so.2+0x9832a)
    #1 0x473293 in free_parsed_cmdline /home/honggyu/work/uftrace/utils/utils.c:796
    #2 0x40ae2b in main /home/honggyu/work/uftrace/uftrace.c:1105
    #3 0x7f06f5c4582f in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x2082f)
    #4 0x405ac8 in _start (/home/honggyu/work/uftrace/uftrace+0x405ac8)

0x60300000efc0 is located 16 bytes inside of 23-byte region [0x60300000efb0,0x60300000efc7)
allocated by thread T0 here:
    #0 0x7f06f72bf34f in strdup (/usr/lib/x86_64-linux-gnu/libasan.so.2+0x6234f)
    #1 0x4730e0 in parse_cmdline /home/honggyu/work/uftrace/utils/utils.c:760
    #2 0x409a5a in parse_opt_file /home/honggyu/work/uftrace/uftrace.c:861
    #3 0x40a839 in main /home/honggyu/work/uftrace/uftrace.c:1010
    #4 0x7f06f5c4582f in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x2082f)

SUMMARY: AddressSanitizer: bad-free ??:0 __interceptor_free
==57330==ABORTING
```

**Comment by honggyukim** [comment]
It seems that the below change fixes it, but I'm not sure if this is enough fix.
```diff
diff --git a/uftrace.c b/uftrace.c
index 2e5b79d..8a723a0 100644
--- a/uftrace.c
+++ b/uftrace.c
@@ -1102,7 +1102,7 @@ int main(int argc, char *argv[])
                fclose(logfp);

        if (opts.opt_file)
-               free_parsed_cmdline(argv);
+               free_parsed_cmdline(argv - opts.idx);

        free_opts(&opts);
        return ret;
```
It works fine anyway.
```
$ ./uftrace -L. --opt-file opts
# DURATION     TID     FUNCTION
   1.940 us [ 58155] | __monstartup();
   1.153 us [ 58155] | __cxa_atexit();
            [ 58155] | main() {
            [ 58155] |   a() {
   2.120 us [ 58155] |     b();
   3.660 us [ 58155] |   } /* a */
   4.600 us [ 58155] | } /* main */
```

**Comment by namhyung** [comment]
Yep, looks like a correct fix.  Please send a PR.

**Comment by honggyukim** [comment]
Just uploaded the fix in #712. Thanks!
